animations/lib/src/misc/carousel.dart

```diff
@@ -0,0 +1,103 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/foundation.dart';
+import 'package:flutter/widgets.dart';
+
+/// This sample is based on the `coverflow` package
+/// https://github.com/efortuna/coverflow
+class CarouselDemo extends StatelessWidget {
+  static String routeName = '/misc/carousel';
+
+  static List<String> fileNames = [
+    'assets/eat_cape_town_sm.jpg',
+    'assets/eat_new_orleans_sm.jpg',
+    'assets/eat_sydney_sm.jpg',
+  ];
+
+  final List<Widget> data = [
+    ...fileNames.map((file) => Image.asset(file, fit: BoxFit.cover))
+  ];
+
+  @override
+  Widget build(context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('Carousel Demo'),
+      ),
+      body: Carousel(itemBuilder: widgetBuilder),
+    );
+  }
+
+  Widget widgetBuilder(context, int index) {
+    return data[index % data.length];
+  }
+}
+
+typedef void OnCurrentItemChangedCallback(int currentItem);
+
+class Carousel extends StatefulWidget {
+  final IndexedWidgetBuilder itemBuilder;
+
+  const Carousel({Key key, @required this.itemBuilder});
+
+  @override
+  _CarouselState createState() => _CarouselState();
+}
+
+class _CarouselState extends State<Carousel> {
+  final _height = 525;
+  final _width = 700;
+  PageController _controller;
+  int _currentPage;
+  bool _pageHasChanged = false;
+
+  @override
+  initState() {
+    super.initState();
+    _currentPage = 0;
+    _controller = PageController(
+      viewportFraction: .85,
+      initialPage: _currentPage,
+    );
+  }
+
+  @override
+  Widget build(context) {
+    return PageView.builder(
+        onPageChanged: (value) {
+          setState(() {
+            _pageHasChanged = true;
+            _currentPage = value;
+          });
+        },
+        controller: _controller,
+        itemBuilder: (context, index) => builder(index));
+  }
+
+  Widget builder(int index) {
+    return AnimatedBuilder(
```

Comment:
optional nit: consider removing this extra builder method and just moving the turn statement up into the above `build`. There are a lot of things named "builder" in this file, and having one fewer could help readers.

Reply:
I'm not sure what you mean -the reason this uses an AnimatedBuilder is to automatically run the `builder` method when the value of the `_controller` changes.

Reply:
I can see why this was unclear. It's not the AnimatedBuilder I was suggesting you remove. It was the `builder` method declared on line 85. Since it's essentially just one return statement, you could simply inline that statement beginning on line 81 and have one less thing in this class with build/builder in its name.

This is a style suggestion for readability's sake, so you're free to ignore it if you feel the code is better as-is.